### PR TITLE
[IoT] Document instructions for use of custom authorizers in IoT

### PR DESCRIPTION
**Pull request by desokroshan**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: android/pubsub.md [hunk 1]
```diff
@@ -139,6 +139,47 @@ try {
...
+        try {
+            String tokenKeyName = <TOKEN_KEY_NAME>;
+            String token = <TOKEN>;
+            String tokenSignature = <TOKEN_SIGNATURE>;
+            String customAuthorizerName = <AUTHORIZER_NAME>;
+            mqttManager.connect(
+	            tokenKeyName,
+                    token,
+		    tokenSignature,
+		    customAuthorizerName,
```
Comment (raphkim) [inline android/pubsub.md]: `tokenKeyName`, `tokenSignature`, and `customAuthorizerName` all seem to have 7 spaces instead of 8

File: android/pubsub.md [hunk 2]
```diff
@@ -139,6 +139,47 @@ try {
...
+                                }
+                            });
+                        }
+                    });
+        } catch (final Exception e) {
+            Log.e(LOG_TAG, "Connection error.", e);
+        }
+```
+
+This feature is available in the AWS SDK for iOS starting from `2.16.48` version. See [AWSIoT - 2.16.8](https://github.com/aws/aws-sdk-android/releases/tag/release_v2.16.8) for more details.
```
Comment (raphkim) [inline android/pubsub.md]: `2.16.48` -> `2.16.8`

File: android/pubsub.md [hunk 3]
```diff
@@ -139,6 +139,47 @@ try {
...
+#### Custom Authentication
+
+AWS IoT allows you to define custom authorizers that allow you to manage your own authentication and authorization strategy using a custom authentication service and a Lambda function. Custom authorizers allow AWS IoT to authenticate your devices and authorize operations using bearer token authentication and authorization strategies. See [AWS IoT Custom Authentication](https://docs.aws.amazon.com/iot/latest/developerguide/custom-authentication.html) for more details.
+
+Please follow the steps outlined in [Setting up Custom Authentication](https://aws.amazon.com/blogs/security/how-to-use-your-own-identity-and-access-management-systems-to-control-access-to-aws-iot-resources/) to create the custom authorizer and configure the workflow with AWS IoT.
+
+Once the custom authorizer workflow is configured, you can establish a connection as follows:
+
+```java
+        try {
```
Comment (raphkim) [inline android/pubsub.md]: can we remove the unnecessary whitespace

File: android/pubsub.md (hunk 1 above)
Comment (desokroshan) [reply android/pubsub.md]: Ah, it was actually tabs instead of spaces. Fixed.

File: android/pubsub.md (hunk 2 above)
Comment (jamesonwilliams) [reply android/pubsub.md]: Nice catch
